File: apps/android/app/src/main/java/pro/wewed/app/models/Wedding.kt
package pro.wewed.app.models

data class ProgrammeItem(
    val id: String,
    val title: String,
    val time: String,
    val location: String,
    val description: String
)

data class Wedding(
    val id: String,
    val coupleNames: String,
    val date: String,
    val venueName: String,
    val venueAddress: String,
    val city: String,
    val country: String,
    val lifecycle: String,
    val programme: List<ProgrammeItem>,
    val mapsUrl: String? = null,
    val latitude: Double? = null,
    val longitude: Double? = null
) {
    val venueLocation: VenueLocation
        get() = VenueLocation(
            name = venueName,
            streetAddress = null,
            city = city,
            country = country,
            mapsUrl = mapsUrl,
            latitude = latitude,
            longitude = longitude
        )
}

/** A venue exactly as recorded on the wedding. streetAddress stays null unless a real street address exists. */
data class VenueLocation(
    val name: String,
    val streetAddress: String? = null,
    val city: String? = null,
    val country: String? = null,
    val mapsUrl: String? = null,
    val latitude: Double? = null,
    val longitude: Double? = null
)

File: apps/ios/Wewed/Models/Wedding.swift
import Foundation

public struct ProgrammeItem: Identifiable, Codable, Equatable, Sendable {
    public let id: String
    public let title: String
    public let time: String
    public let location: String
    public let description: String

    public init(id: String, title: String, time: String, location: String, description: String) {
        self.id = id
        self.title = title
        self.time = time
        self.location = location
        self.description = description
    }
}

public struct Wedding: Identifiable, Codable, Equatable, Sendable {
    public let id: String
    public let coupleNames: String
    public let date: String
    public let venueName: String
    public let venueAddress: String
    public let city: String
    public let country: String
    public let lifecycle: String
    public let programme: [ProgrammeItem]

    public init(
        id: String,
        coupleNames: String,
        date: String,
        venueName: String,
        venueAddress: String,
        city: String,
        country: String,
        lifecycle: String,
        programme: [ProgrammeItem]
    ) {
        self.id = id
        self.coupleNames = coupleNames
        self.date = date
        self.venueName = venueName
        self.venueAddress = venueAddress
        self.city = city
        self.country = country
        self.lifecycle = lifecycle
        self.programme = programme
    }
}
